.editorconfig

```diff
@@ -4,6 +4,45 @@
 
 # Please keep in alphabetical order by field.
 
+# We cannot reasonably mark ourselves CLSCompliant one way or another if
```

Comment:
You mean like a developer disabling rules on their workstation that we'd want on for the repo? No, these rules apply to the code itself and need to match for everyone so they can know what we/CI expect.

Locally I go and comment out `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` in `Directory.Build.props` sometimes while I work just so I can tell compile failures from these static analyzer ones.

Reply:
I meant locally as in per location exception of these rules.

Reply:
Ah, you'll see I do that with pragmas sometimes. These I finally disable are specifically because I disagree with them at the project level for all code. For cases where I need local overrides (i.e. it's a good rule but I have to violate for some special reason), I do that very specifically. I comment here and in those places and can provide details on any specific rule decision if needed.